ci/jenkins/test-vm.sh

```diff
@@ -139,26 +142,66 @@ function apply_antrea {
     kubectl apply -f ${WORKDIR}/antrea.yml
 }
 
-function clean_vm_agent {
-    echo "====== Cleanup Antrea-Agent Installation on the VM ======"
-    echo "Revert to snapshot VMAgent"
-    govc snapshot.revert -k=true -vm=${LIN_HOSTNAME} VMAgent
-    govc snapshot.revert -k=true -vm=${WIN_HOSTNAME} VMAgent
-    echo "Get IP addresses for VMs"
-    UBUNTU_VM_IP=$(govc vm.ip -k=true -wait=1m ${LIN_HOSTNAME})
-    for i in `seq 10`; do
-        WINDOWS_VM_IP=$(govc vm.ip -k=true -wait=2m ${WIN_HOSTNAME})
-        if [[ WINDOWS_VM_IP == "" ]]; then
-            echo "Failed to retrieve IP for Windows VM ${WIN_HOSTNAME}, retry ${i}"
-            continue
-        fi
-    done
+function export_govc_env_var {
+    # This should be coming from jenkins configuration
+    export GOVC_URL=$GOVC_URL
+    export GOVC_USERNAME=$GOVC_USERNAME
+    export GOVC_PASSWORD=$GOVC_PASSWORD
+    export GOVC_INSECURE=1
+    export GOVC_DATACENTER=$GOVC_DATACENTER
+    export GOVC_DATASTORE=$GOVC_DATASTORE
+}
+
+function clean_up_one_ns {
+    ns=$1
+    kubectl delete ns "${ns}" --ignore-not-found=true || true
+}
 
+function clean_service_account_config {
     kubectl delete sa $SERVICE_ACCOUNT -n $TEST_NAMESPACE --ignore-not-found=true
     clean_up_one_ns $TEST_NAMESPACE
-    echo "Deleting stale antrea-agent files from ${LIN_HOSTNAME} and ${WIN_HOSTNAME}"
-    ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ubuntu@${UBUNTU_VM_IP} "rm -rf /tmp/antrea-ci"
-    ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" Administrator@${WINDOWS_VM_IP} "rm -rf /tmp/antrea-ci"
+}
+
+function clean_vm_agent_on_linux {
+    for host_name in "${!LINUX_HOSTS_TO_IP[@]}"; do
+      echo "====== Cleanup Antrea-Agent Installation on the $host_name VM ======"
+      echo "Revert to snapshot InstallVMAgent"
+      USERNAME=${LINUX_HOSTS_TO_USERNAME[${host_name}]}
+      IP_ADDRESS=${LINUX_HOSTS_TO_IP[${host_name}]}
+      govc snapshot.revert -k=true -vm=$host_name InstallVMAgent
+      echo "Deleting stale antrea-agent files from $host_name"
+      ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" -n ${USERNAME}@${IP_ADDRESS} "rm -rf /tmp/antrea-ci"
+    done
+}
+
+function clean_vm_agent_on_windows {
+    for host_name in "${!WINDOWS_HOSTS_TO_IP[@]}"; do
+      echo "====== Cleanup Antrea-Agent Installation on the $host_name VM ======"
+      echo "Revert to snapshot InstallVMAgent"
+      USERNAME=${WINDOWS_HOSTS_TO_USERNAME[${host_name}]}
+      IP_ADDRESS=${WINDOWS_HOSTS_TO_IP[${host_name}]}
+      govc snapshot.revert -k=true -vm=$host_name InstallVMAgent
+      echo "Deleting stale antrea-agent files from $host_name"
+      ssh -o StrictHostKeyChecking=no -i "${WORKDIR}/jenkins_id_rsa" ${USERNAME}@${IP_ADDRESS} "rm -rf /tmp/antrea-ci"
```

Comment:
Because we are creating that directory manually even if we use different name later we can directly delete it easily or else we should modify the snapshot.